core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -288,4 +304,43 @@ class SpendReportingService(
     }
   }
 
+  def getSpendForBillingProject(
+    project: RawlsBillingProjectName,
+    start: DateTime,
+    end: DateTime,
+    aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] =
+    for {
+      billingProject <- billingRepository.getBillingProject(project)
+
+      report <- getReportData(billingProject.get, project, start, end, aggregations)
+    } yield report
+
+  private def getReportData(billingProject: RawlsBillingProject,
+                            project: RawlsBillingProjectName,
+                            start: DateTime,
+                            end: DateTime,
+                            aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] =
+    if (billingProject.billingProfileId.isEmpty) {
+      getSpendForGCPBillingProject(project, start, end, aggregations)
+    } else {
+      getSpendForAzureBillingProject(billingProject.billingProfileId.get, start, end)
+    }
+
+  private def getSpendForAzureBillingProject(
+    billingProfileId: String,
+    start: DateTime,
+    end: DateTime
+  ): Future[SpendReportingResults] =
```

Comment:
@blakery  and I chatted. This could just return `SpendReportingResults`, and the call could be wrapped with a Future where it's actually called (line 328).

As the method is currently written (with return type of Future[SpendReportingResults], it would be better to return Future.failed(exception) than to throw the exception outside of a Future.